### ISSUE TITLE
Incompatibility with symfony/event-dispatcher

### ISSUE THREAD

**Issue by zebraf1** [php-vcr/php-vcr#336]
Using php-vcr 1.5 conflicts with symfony/event-dispatcher 5.0

requires symfony/event-dispatcher: ^2.4|^3.0|^4.0|^5.0

** How to reproduce: **
* install php-vcr ^1.5
* install symfony/event-dispatcher ^5.0

 
[TypeError] Argument 1 passed to Symfony\Component\EventDispatcher\EventDispatcher::dispatch() must be an object, string given, called in .../vendor/php-vcr/php-vcr/src/VCR/Videorecorder.php on line 99

* https://github.com/symfony/event-dispatcher/blob/5.0/CHANGELOG.md#430 (dispatch method signature changed and raises deprecation warning)
* https://github.com/symfony/event-dispatcher/blob/5.0/CHANGELOG.md#500
The signature of the EventDispatcherInterface::dispatch() method has been changed to dispatch($event, string $eventName = null): object.

**Problem:**
dispatch is called with (string, Event) types.
https://github.com/symfony/event-dispatcher/blob/5.0/EventDispatcher.php#L48 - this new interface requires first parameter to be object.

**Solution**:
* Use appropriate way to check whether EventDispatcher is using old or new interface
* Do not rely on class_exists(\Symfony\Component\EventDispatcher\Event::class)

**Comment by zebraf1** [comment]
https://github.com/php-vcr/php-vcr/commit/3d88f89ae2f915471043ebe61ff964937bc246d6#diff-8f9fb128fc0a304a8c6dcd7ba61771a13170c368a48a9fabf993fc097416469fR104

This code should make the code compatible with 5.0 also. So it is weird that my project finds \Symfony\Component\EventDispatcher\Event::class exists

**Comment by zebraf1** [comment]
If you use codeception/codeception ^4.1.15 then you will have Event::class at that namespace due to this change:

https://github.com/Codeception/Codeception/commit/15ba342b6511d0557435182d815690aa6b8e6aef
```php
//Compatibility with Symfony 5
if (!class_exists('Symfony\Component\EventDispatcher\Event') && class_exists('Symfony\Contracts\EventDispatcher\Event')) {
    class_alias('Symfony\Contracts\EventDispatcher\Event', 'Symfony\Component\EventDispatcher\Event');
}
```

**Comment by zebraf1** [comment]
So the problem is that @pauladams8 assumed that if Event class is missing then project is using the new interface For EventDispatcher, but @natikbalda added an alias for this class in Codeception for bc.

So you cannot use Codeception/codeception 4.1.15 with php-vcr 1.5 unless you restrict a package from upgrading in composer.json:

"symfony/event-dispatcher": "<5.0"
OR
"codeception/codeception": "<4.1.15"
OR
"php-vcr/php-vcr": "<1.5"

and run composer update

**Comment by zebraf1** [comment]
Of course changing an interface in such a way is bound to cause problems in other libraries. Tsk tsk.

**Comment by Naktibalda** [comment]
This issue will be fixed in Codeception 5.0.0 in February.

**Comment by Naktibalda** [comment]
The best way to check if EventDispatcher supports new interface was implemented by https://github.com/Codeception/Codeception/pull/6066/files

```php
<?php
use Symfony\Contracts\EventDispatcher\EventDispatcherInterface as ContractsEventDispatcherInterface;

// The `EventDispatcherInterface` of `Symfony\Contracts` is only implemented in Symfony 4.3 or higher
if ($dispatcher instanceof ContractsEventDispatcherInterface) {
	//Symfony 4.3 or higher
	$dispatcher->dispatch($eventObject, $eventType);
} else {
	//Symfony 4.2 or lower
	$dispatcher->dispatch($eventType, $eventObject);
}
```

**Comment by federicoemartinez** [comment]
Hi!
We are facing the same issue. Will this be released?
